mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -437,15 +437,15 @@ def MIOpen_BlockwiseGemmOp:
 // blockwise_gemm_v2
 def MIOpen_BlockwiseGemmV2Op:
     MIOpen_Op<"blockwise_gemm_v2">,
-    Arguments<(ins MemRefOf<[F32, F16, BF16]>:$matrixA,
-                   MemRefOf<[F32, F16, BF16]>:$matrixB,
+    Arguments<(ins MemRefOf<[F32, F16, BF16, I8]>:$matrixA,
+                   MemRefOf<[F32, F16, BF16, I8]>:$matrixB,
                    ArgTransforms<2>:$transforms,
                    Index:$waveOffsetA,
                    Index:$waveOffsetB,
-                   MemRefOf<[F32, F16, BF16, VectorOfLengthAndType<[2, 4], [F32]>, VectorOfLengthAndType<[2, 4, 8], [F16, BF16]>]>:$bufferA,
-                   MemRefOf<[F32, F16, BF16, VectorOfLengthAndType<[2, 4], [F32]>, VectorOfLengthAndType<[2, 4, 8], [F16, BF16]>]>:$bufferB,
-                   Variadic<VectorOfRankAndType<[1], [F32, F16]>>:$vectorCs)>,
-    Results<(outs Variadic<VectorOfRankAndType<[1], [F32, F16]>>: $vectorDs)> {
+                   MemRefOf<[F32, F16, BF16, I8, VectorOfLengthAndType<[2, 4], [F32]>, VectorOfLengthAndType<[2, 4, 8], [F16, BF16]>, VectorOfLengthAndType<[4], [I8]>]>:$bufferA,
```

Comment:
Shouldn't those VectorOfLengthAndType<[4], [I8]> be something like [4, 16], or am I misremembering the context?

Also, while you're here, could you break those constraints up onto multiple lines - they've gotten really long.

Reply:
Sure, will break to make it shorter.

The reason for `VectorOfLengthAndType<[4], [I8]>` definition is that for both xdlops exposed they are only able to execute the argument of `i32`, or equivalent to 4xi8.

In the lowered code, bufferA and bufferB will load temporary result and get directly feed into the xdlops instruction therefore it has to match the length of argument with xdlops.

Reply:
Noted that both the xdlops just take i32 - for some reason, I thought some of them were wider

Reply:
Agreed, I think you meant to say that the data preparation time we can take advantage of kpack to better take advantage of the bandwidth, which will translate to more inner loops in xdlops execution.

Reply:
breaking lines addressed in [700a3a9](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/505/commits/700a3a9c5b5a3522b38d16dfc9cb77cfa984018b)